tsfile/src/main/java/org/apache/iotdb/tsfile/read/reader/page/TimePageReader.java

```diff
@@ -53,12 +54,11 @@ public TimePageReader(PageHeader pageHeader, ByteBuffer pageData, Decoder timeDe
   }
 
   public long[] nexTimeBatch() throws IOException {
-    long[] timeBatch = new long[(int) pageHeader.getStatistics().getCount()];
-    int index = 0;
+    List<Long> timeList = new ArrayList<>();
     while (timeDecoder.hasNext(timeBuffer)) {
-      timeBatch[index++] = timeDecoder.readLong(timeBuffer);
+      timeList.add(timeDecoder.readLong(timeBuffer));
     }
-    return timeBatch;
+    return timeList.stream().mapToLong(t -> t.longValue()).toArray();
```

Comment:
Why is this modification necessary? This modification will cause performance loss, which is equivalent to two iterations. o(n*n)